ui/components/src/test/propertygrid/dataproviders/filterers/CompositePropertyDataFilterer.test.ts

```diff
@@ -90,63 +93,123 @@ describe("CompositePropertyDataFilterer", () => {
     });
 
     describe("matchesFilter", () => {
```

Comment:
Tests in here are duplicated for records and categories. Try to avoid duplication by extracting the common pieces and reusing them.